docs/rfcs/rfc-020-array-methods.md

```diff
@@ -0,0 +1,86 @@
+# Array Methods
+
+## Problem
+
+Our users would like to modify arrays inline.
+
+## Proposed Solution
+
+Add new array methods PUSH and POP:
+
+-   `_PUSH`
+-   `_POP`
+
+### Potential additional methods
+
+SHIFT `_SHIFT` and UNSHIFT `_UNSHIFT` are very similar methods to PUSH and POP so they may be included, if time permits.
+
+### Usage Examples
+
+Given the following type definitions with a middleNames array of strings:
+
+```gql
+type Person {
+    name: String!
+    middleNames: [String]
+}
+```
+
+We could perform a POP operation like so, where the update parameter is an integer, specifying the number of array elements to pop off the end of the array. Popped off elements would then be returned in `poppedElements` inside the `info` object.
+
+The minimum valid value is 1. Values less than 1 would result in an error.
+
+```
+mutation {
+  updateRecord(
+    where: { name: "Some Person" }
+    update: { middleNames_POP: 1 }
+  ) {
+    records {
+      name
+      middleNames
+    }
+    info {
+      poppedElements: ["Emil"]
+    }
+  }
+}
+```
+
+We could perform a PUSH operation like so, where the update parameter is an array containing the values to be pushed to the end of the array:
+
+```
+mutation {
+  updateRecord(
+    where: { name: "Some Person" }
+    update: { middleNames_PUSH: ["Emil"] }
+  ) {
+    records {
+      name
+      middleNames
+    }
+  }
+}
+```
+
+### Technical considerations
+
+In an update mutation when the POP method is not used, the `info` object will contain the field `poppedElements` with a value of `[]`.
+
```

Comment:
A couple of things that I would add as technical considerations are:

- What happens when used on an optional/not defined field.
- What happens when the same query contains another operator/setter for the same field(I would suggest raising an error there!)

Reply:
I think I've now addressed your points here @MacondoExpress. If you agree I'll go ahead and resolve this comment.

Reply:
Absolutely! good joob!